From 1179367e22b02acb6557a082e470d765aa344ff1 Mon Sep 17 00:00:00 2001
From: Poovendhan Selvaraj <quic_poovendh@quicinc.com>
Date: Wed, 14 Aug 2024 17:20:47 +0530
Subject: [PATCH] drivers: qcom_scm.: Add fix to handle dma unmap

Added fix to handle dma unmap before returning from seccrypt function

Signed-off-by: Poovendhan Selvaraj <quic_poovendh@quicinc.com>
Change-Id: I5a4c55c6d7b64eaf314dc3f653a155bde4925268
---
 drivers/firmware/qcom_scm.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/firmware/qcom_scm.c b/drivers/firmware/qcom_scm.c
index 364e570f4189..6a03404bf3ac 100644
--- a/drivers/firmware/qcom_scm.c
+++ b/drivers/firmware/qcom_scm.c
@@ -1846,7 +1846,7 @@ int __qti_sec_crypt(struct device *dev, void *confBuf, int size)
 	desc.args[0] = (u64)conf_phys;
 	desc.args[1] = size;
 
-	return qcom_scm_call(__scm->dev, &desc, &res);
+	ret = qcom_scm_call(__scm->dev, &desc, &res);
 
 	dma_unmap_single(dev, conf_phys, size, DMA_TO_DEVICE);
 	return ret ? : res.result[0];
-- 
2.34.1

